### ISSUE TITLE
1.9.0 Deno.connect() types the connection to Conn<NetAddr> making code incompatible

### ISSUE THREAD

**Issue by aricart** [denoland/deno#10200]
Previous versions of deno had non-generic `Conn`. This breaks libraries that work perfectly with any version of deno.

**Comment by aricart** [comment]
Obviously, the backward and forwards solution is ``// @ts-ignore: Deno added generics to Conn` but maybe there's something that can be done that is more elegant.

**Comment by bartlomieju** [comment]
@aricart could you provide example of code that got broken?

**Comment by aricart** [comment]
Sorry - Deno.startTls changed signature, so code as below, will fail to compile in 1.9.0

```
let conn: Deno.Conn;
const hostname = "connect.ngs.global";

conn = await Deno.connect({hostname: hostname, port: 4222});
conn = await Deno.startTls(
  conn,
  { hostname },
);

console.log("connected");
```

**Comment by aricart** [comment]
```bash
error: TS2345 [ERROR]: Argument of type 'Conn<Addr>' is not assignable to parameter of type 'Conn<NetAddr>'.
  Type 'Addr' is not assignable to type 'NetAddr'.
    Type 'UnixAddr' is missing the following properties from type 'NetAddr': hostname, port
  conn,


Pacifying it on 1.9.0 by specifying the `Conn<

**Comment by kitsonk** [comment]
Doing this:

```ts
const hostname = "connect.ngs.global";

let conn = await Deno.connect({ hostname: hostname, port: 4222 });
// other stuff happening here after detecting the server supports TLS

conn = await Deno.startTls(
  conn,
  { hostname: hostname },
);

console.log("connected");
```

Fixes it under Deno 1.9.  Is that not acceptable?

**Comment by kt3k** [comment]
I think #10012 wasn't breaking with the above example if we had set default of type parameter to `NetAddr` like:

```
interface Conn<Address extends Addr = NetAddr>
```
